tests/test_rotations.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+import pytest
+from numpy.typing import NDArray
+from scipy.spatial.transform import Rotation
+
+from timemachine.lib import custom_ops
+
+
+def hamiliton_product(q1: NDArray, q2: NDArray) -> NDArray:
+    """
+    Compute the product of two quaternions, called a hamilton product.
+
+    References
+    ----------
+    https://en.wikipedia.org/wiki/Quaternion#Hamilton_product
+    """
+    w1, x1, y1, z1 = q1[0], q1[1], q1[2], q1[3]
```

Comment:
nit: could this just be
```suggestion
    w1, x1, y1, z1 = q1
```
?

Reply:
Yup, will change

Reply:
Changed in https://github.com/proteneer/timemachine/pull/1174/commits/ff88f8762f5e809c6df5e09b36a5ac85b268ad04